### PR TITLE
Auto log the stats in the new bulk executor

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
The auto logging of stats feature is missing in the new bulk execution backend (https://github.com/ray-project/ray/pull/30903). This PR added it.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

**Comment by c21** [comment]
@scottjlee - can you help review this PR? Thanks.

**Review by scottjlee — APPROVED** [review]

File: python/ray/data/_internal/execution/bulk_executor.py [hunk 1]
```diff
@@ -35,7 +36,7 @@ def execute(
         assert not self._executed, "Can only call execute once."
         self._executed = True
         if not isinstance(dag, InputDataBuffer):
-            logger.info("Executing DAG %s", dag)
+            logger.get_logger().info("Executing DAG %s", dag)
```
Comment (scottjlee) [inline python/ray/data/_internal/execution/bulk_executor.py:39]: Wanted to confirm that this and the `debug()` a few lines below are intended to always write to both dataset logfile and stdout, as opposed to using `log_to_stdout=context.enable_auto_log_stats` to conform with the context configuration.

Comment (jianoaix) [reply python/ray/data/_internal/execution/bulk_executor.py:39]: This message is not stats so I think `enable_auto_log_stats` doesn't apply it.

**Comment by jianoaix** [comment]
@ericl @clarkzinzow Review or merge? Thanks.

**Pull request merged**